File: tests/rust/tests-rust-testing-reqwest-selenium/tests/ui/burger_menu_tests.rs
use allure_cargotest::allure_test;
use tests_rust_testing_reqwest_selenium as tests;

fn layer() {
    tests::layer_ui("Burger menu", "Header", "Burger menu", "normal");
}

#[allure_test(name = "Menu nav marks Login active on the login page")]
#[tokio::test]
async fn menu_nav_marks_active_login() {
    layer();
    tests::with_browser(|| async {
        tests::LoginPage::default()
            .header
            .set_mobile_viewport()
            .await;
        tests::LoginPage::default()
            .open_page()
            .await
            .header
            .open_menu()
            .await
            .should_have_active_menu_nav("header-menu-nav-login")
            .await;
        tests::LoginPage::default().header.reset_viewport().await;
    })
    .await;
}

#[allure_test(name = "Menu Register opens the register page and closes the menu")]
#[tokio::test]
async fn clicking_register_opens_register_and_closes_menu() {
    layer();
    tests::with_browser(|| async {
        tests::LoginPage::default()
            .header
            .set_mobile_viewport()
            .await;
        tests::LoginPage::default()
            .open_page()
            .await
            .header
            .open_menu()
            .await
            .should_have_active_menu_nav("header-menu-nav-login")
            .await
            .click_menu_nav("header-menu-nav-register")
            .await;
        tests::RegisterPage::default()
            .should_be_open()
            .await
            .header
            .should_have_closed_menu()
            .await;
        tests::LoginPage::default().header.reset_viewport().await;
    })
    .await;
}

#[allure_test(name = "Menu Login opens the login page and closes the menu")]
#[tokio::test]
async fn clicking_login_opens_login_and_closes_menu() {
    layer();
    tests::with_browser(|| async {
        tests::LoginPage::default()
            .header
            .set_mobile_viewport()
            .await;
        tests::RegisterPage::default()
            .open_page()
            .await
            .header
            .open_menu()
            .await
            .click_menu_nav("header-menu-nav-login")
            .await;
        tests::LoginPage::default()
            .should_be_open()
            .await
            .header
            .should_have_closed_menu()
            .await;
        tests::LoginPage::default().header.reset_viewport().await;
    })
    .await;
}

File: tests/kotlin/tests-kotlin-junit5-ktor-selenium/src/test/kotlin/tests/ui/BurgerMenuTests.kt
package tests.ui

import tests.TestBase
import annotations.Layer
import io.qameta.allure.Epic
import io.qameta.allure.Feature
import io.qameta.allure.Severity
import io.qameta.allure.SeverityLevel
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test

@Layer("ui")
@Epic("Header")
@Feature("Burger menu")
@Severity(SeverityLevel.NORMAL)
@DisplayName("Burger menu")
class BurgerMenuTests : TestBase() {

    @BeforeEach
    fun setMobileViewport() {
        loginPage.header.setMobileViewport()
    }

    @AfterEach
    fun resetViewport() {
        loginPage.header.resetViewport()
    }

    @Test
    @Tag("ui")
    @DisplayName("Menu nav marks Login active on the login page")
    fun menuNavMarksActiveLogin() {
        loginPage.openPage()
                .header.openMenu()
                .shouldHaveActiveMenuNav("header-menu-nav-login")
    }

    @Test
    @Tag("ui")
    @DisplayName("Menu Register opens the register page and closes the menu")
    fun clickingRegisterOpensRegisterAndClosesMenu() {
        loginPage.openPage()
                .header.openMenu()
                .shouldHaveActiveMenuNav("header-menu-nav-login")
                .clickMenuNav("header-menu-nav-register")
        registerPage.shouldBeOpen()
                .header.shouldHaveClosedMenu()
    }

    @Test
    @Tag("ui")
    @DisplayName("Menu Login opens the login page and closes the menu")
    fun clickingLoginOpensLoginAndClosesMenu() {
        registerPage.openPage()
                .header.openMenu()
                .clickMenuNav("header-menu-nav-login")
        loginPage.shouldBeOpen()
                .header.shouldHaveClosedMenu()
    }
}
